keps/sig-node/4815-dra-partitionable-devices/README.md

```diff
@@ -0,0 +1,2437 @@
+|          2g.10gb        |          2g.10gb        |          2g.10gb        |            X            |
++------------+------------+------------+------------+------------+------------+------------+------------+
+|          1g.10gb        |          1g.10gb        |          1g.10gb        |          1g.10gb        |
++------------+------------+------------+------------+------------+------------+------------+------------+
+|   1g.5gb   |   1g.5gb   |   1g.5gb   |   1g.5gb   |   1g.5gb   |   1g.5gb   |   1g.5gb   |     X      |
++------------+------------+------------+------------+------------+------------+------------+------------+
+| 1g.5gb+me  | 1g.5gb+me  | 1g.5gb+me  | 1g.5gb+me  | 1g.5gb+me  | 1g.5gb+me  | 1g.5gb+me  |     X      |
++------------+------------+------------+------------+------------+------------+------------+------------+
+```
+
+The most important thing to note from all of this is that partitioning a GPU
+into a set of MIG devices is not as simple as slicing it up into a set of
+fixed-size, non-overlapping partitions. Instead, one can think of a GPU as a
+"bag" of resources that can be pulled together in a discrete set of ways to
+construct a MIG device with a particular profile. When instatiating a MIG
+device with a particular profile, the set of resources available to other MIG
+devices (or the full GPU itself) gets depleted.
+
+In other words, the allocation of one MIG device may nullify the
+ability to allocate some subset of other possible MIG devices (but not all of
+them). A MIG device is considered allocatable so long as the full GPU has
+enough *unallocated* capacity to satisfy its capacity constraints across all
+dimensions. If any single dimension is unavailable, that MIG device cannot be
+allocated (but others might still be able to).
+
+With his motivating example in mind. We define the following goals and
+non-goals of this KEP.
+
+### Goals
+
+* Introduce the ability for "structured parameters" DRA to allocate both full
+  devices and fixed-size partitions of full devices (across multiple
+  dimensions). Both full devices and their valid set of partitions must be
+  explicitly listed in a ResourceSlice for the scheduler to consider them for
+  allocation.
+
+* Provide abstractions to keep the overall footprint of a ResourceSlice small,
+  such that all full devices (and their partitions) can be represented in a
+  concise way.
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+* Allow a user to allocate arbitrarily-sized partitions of a device. In other
+  words, only those partitions explicitly defined by a ResourceSlice will be
+  allocatable by a user. Arbitrary slicing is out of scope for this KEP.
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
```

Comment:
While reading this, this KEP seems to focus heavily on the device vendor needs for ResourceSlices changes for Partitionable Devices. You do not mention any user changes for this.

If I am user and I want to use particular MIG slices, what the UX experience look like with this feature?

I assume I create a resource claim that points to a device but does the user have the ability to make sure they get certain MIG slices?

Reply:
I guess that's because there are no user-facing changes. From the perspective of a user they still request devices exactly as they did before -- as a CEL expression with a boolean expression over its attributes and capacities. Maybe I should just state that explicitly somewhere.

Reply:
I think it will be good to add that example :+1:

Reply:
I have added this example in the text.